gradle-sls-packaging/src/main/java/com/palantir/gradle/dist/service/JavaServiceDistributionPlugin.java

```diff
@@ -264,6 +268,21 @@ public void execute(Task _task) {
             } else {
                 task.getMainClass().set(mainClassName);
             }
+            task.doFirst(new Action<Task>() {
```

Comment:
I think we can avoid the `doFirst` by using https://docs.gradle.org/current/javadoc/org/gradle/process/JavaForkOptions.html#getJvmArgumentProviders--